internal/consensus/state.go

```diff
@@ -2244,6 +2325,10 @@ func (cs *State) defaultSetProposal(proposal *types.Proposal, recvTime time.Time
 		return ErrProposalTooManyParts
 	}
 
+	if int64(proposal.BlobID.PartSetHeader.Total) > (maxBytes-1)/int64(types.BlockPartSizeBytes)+1 {
```

Comment:
I could be wrong, but this could be?
```suggestion
	if int64(proposal.BlobID.PartSetHeader.Total) > (maxBytes-1)/int64(types.BlobPartSizeBytes)+1 {
```